tools/e2e-commons/flows/onboarding.ts

```diff
@@ -0,0 +1,123 @@
+import logger from '../logger';
+import type { Page } from '@playwright/test';
+
+type RedirectToWpcomOptions = {
+	wpcomLoggedIn?: boolean;
+};
+
+// Onboarding API calls and redirects can take a while to complete, so we increase the timeout.
+const DEFAULT_TIMEOUT = 60000;
+
+export class Onboarding {
+	constructor( protected page: Page ) {}
+
+	get CTA() {
+		return this.page.getByRole( 'button', { name: 'Supercharge my site' } );
+	}
+
+	/**
+	 * Start the onboarding process by clicking the CTA button.
+	 *
+	 * @return A promise that resolves when the site connection is completed.
+	 */
+	async start() {
+		logger.info( 'Click on "Supercharge my site" button to start onboarding' );
+
+		await this.CTA.click();
+	}
+
+	/**
+	 * Wait for site connection to complete.
+	 *
+	 * @return A promise that resolves when the site connection is completed.
+	 */
+	async waitForSiteConnection() {
+		logger.info( 'Waiting for site connection to finish.' );
+
+		return await this.page.waitForResponse(
+			response => {
+				return response.url().includes( 'jetpack/v4/connection/register' );
+			},
+			{ timeout: DEFAULT_TIMEOUT }
```

Comment:
I think site connection might be the only one that takes more time than usual. How about setting 60s only here, and leave the default timeout for other queries?

It's just a thought of course

Reply:
I noticed that other redirects sometimes take more time to finish the background process before the redirect actually happens.

Reply:
I'll note that for `waitForURL` (used elsewhere in these tests), the `waitUntil: 'domcontentloaded'` param might be worth using instead of `timeout`.

Reply:
The [default](https://playwright.dev/docs/api/class-page#page-wait-for-url) for that is the `'load'` event but it's still bound by timeout even if you use `'domcontentloaded'`, which means it will wait only for the default timeout which is 20 seconds.

Reply:
Yep, just noting it as I've seen it significantly reduce the wait time (e.g. Automattic/wp-calypso#102743).